File: Colas.cs
using System;

class Cola
{
    const int MAXSIZE = 5;
    static int[] queue = new int[MAXSIZE];
    static int front = -1, rear = -1;

    static void Insertar()
    {
        Console.Write("Ingrese el elemento a insertar: ");
        int elemento = int.Parse(Console.ReadLine());

        if (rear == MAXSIZE - 1)
        {
            Console.WriteLine("Desbordamiento de cola");
            return;
        }

        if (front == -1 && rear == -1)
        {
            front = rear = 0;
        }
        else
        {
            rear++;
        }

        queue[rear] = elemento;
        Console.WriteLine("Elemento insertado: " + elemento);
    }

    static void Eliminar()
    {
        if (front == -1 || front > rear)
        {
            Console.WriteLine("Subdesbordamiento de cola");
            return;
        }

        int elemento = queue[front];

        if (front == rear)
        {
            front = rear = -1;
        }
        else
        {
            front++;
        }

        Console.WriteLine("Elemento eliminado: " + elemento);
    }

    static void Mostrar()
    {
        if (rear == -1 || front == -1 || front > rear)
        {
            Console.WriteLine("La cola está vacía");
        }
        else
        {
            Console.Write("Elementos en la cola: ");
            for (int i = front; i <= rear; i++)
            {
                Console.Write(queue[i] + " ");
            }
            Console.WriteLine();
        }
    }

    static void Main(string[] args)
    {
        int opcion;

        do
        {
            Console.WriteLine("\nMenú de operaciones de cola");
            Console.WriteLine("1. Insertar");
            Console.WriteLine("2. Eliminar");
            Console.WriteLine("3. Mostrar");
            Console.WriteLine("4. Salir");
            Console.Write("Ingrese su opción: ");
            opcion = int.Parse(Console.ReadLine());

            switch (opcion)
            {
                case 1:
                    Insertar();
                    break;

                case 2:
                    Eliminar();
                    break;

                case 3:
                    Mostrar();
                    break;

                case 4:
                    Console.WriteLine("Saliendo del programa");
                    break;

                default:
                    Console.WriteLine("Opción inválida");
                    break;
            }

        } while (opcion != 4);
    }
}

File: Colas.py
MAXSIZE = 5
queue = [0] * MAXSIZE
front = -1
rear = -1

def insertar():
    global front, rear
    elemento = int(input("Ingrese el elemento a insertar: "))

    if rear == MAXSIZE - 1:
        print("Desbordamiento de cola")
        return

    if front == -1 and rear == -1:
        front = rear = 0
    else:
        rear += 1

    queue[rear] = elemento
    print(f"Elemento insertado: {elemento}")

def eliminar():
    global front, rear

    if front == -1 or front > rear:
        print("Subdesbordamiento de cola")
        return

    elemento = queue[front]

    if front == rear:
        front = rear = -1
    else:
        front += 1

    print(f"Elemento eliminado: {elemento}")

def mostrar():
    if rear == -1 or front == -1 or front > rear:
        print("La cola está vacía")
    else:
        print("Elementos en la cola:", end=" ")
        for i in range(front, rear + 1):
            print(queue[i], end=" ")
        print()

def menu():
    while True:
        print("\nMenú de operaciones de cola")
        print("1. Insertar")
        print("2. Eliminar")
        print("3. Mostrar")
        print("4. Salir")

        opcion = input("Ingrese su opción: ")

        match opcion:
            case "1":
                insertar()
            case "2":
                eliminar()
            case "3":
                mostrar()
            case "4":
                print("Saliendo del programa")
                break
            case _:
                print("Opción inválida")

menu()
